pkg/tsdb/cloudmonitoring/time_series_filter.go

```diff
@@ -4,105 +4,21 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
-	"net/http"
 	"net/url"
-	"path"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/grafana/grafana-plugin-sdk-go/backend"
 	"github.com/grafana/grafana-plugin-sdk-go/data"
 	"github.com/huandu/xstrings"
-	"go.opentelemetry.io/otel/attribute"
 
-	"github.com/grafana/grafana/pkg/infra/log"
 	"github.com/grafana/grafana/pkg/infra/tracing"
 )
 
-func doRequestFilterPage(ctx context.Context, r *http.Request, dsInfo datasourceInfo, params url.Values, logger log.Logger) (cloudMonitoringResponse, error) {
-	r.URL.RawQuery = params.Encode()
-	r = r.WithContext(ctx)
-	res, err := dsInfo.services[cloudMonitor].client.Do(r)
-	if err != nil {
-		return cloudMonitoringResponse{}, err
-	}
-
-	dnext, err := unmarshalResponse(logger, res)
-	if err != nil {
-		return cloudMonitoringResponse{}, err
-	}
-
-	return dnext, nil
-}
-
-func runTimeSeriesRequest(ctx context.Context, req *backend.QueryDataRequest,
-	s *Service, dsInfo datasourceInfo, tracer tracing.Tracer, projectName string, params url.Values, logger log.Logger) (*backend.DataResponse, cloudMonitoringResponse, string, error) {
-	dr := &backend.DataResponse{}
-	if projectName == "" {
-		var err error
-		projectName, err = s.getDefaultProject(ctx, dsInfo)
-		if err != nil {
-			dr.Error = err
-			return dr, cloudMonitoringResponse{}, "", nil
-		}
-		logger.Info("No project name set on query, using project name from datasource", "projectName", projectName)
-	}
-	r, err := createRequest(logger, &dsInfo, path.Join("/v3/projects", projectName, "timeSeries"), nil)
-	if err != nil {
-		dr.Error = err
-		return dr, cloudMonitoringResponse{}, "", nil
-	}
-	alignmentPeriod, ok := r.URL.Query()["aggregation.alignmentPeriod"]
```

Comment:
Is this alignment period logic no longer necessary?

Reply:
yes, nicely spotted! I forgot to comment this. This alignment period was added to the custom metadata (line 66) but the whole `Meta` was reset in the `parseTimeSeriesResponse` method (previously line 125) and was not used in the meantime so as far as I can see it was not used.